src/components/TimePicker/TimePicker.js

```diff
@@ -183,64 +200,73 @@ function TimePicker({forwardedRef, defaultValue, onSubmit, onInputChange}) {
+                // The first entered digit is 0-5, we can safely append the second digit.
+                newMinute = `${firstDigit}${trimmedText[2] || 0}`;
                 newSelection = 1;
             } else {
-                newMinute = `${filteredText.slice(0, 2)}`;
+                // The first entered digit is 6-9. We should replace the whole value by prepending 0 to the entered digit.
+                newMinute = `0${firstDigit}`;
                 newSelection = 2;
             }
-        } else if (selectionMinute.start === 0) {
-            // Case when the cursor is at the start.
-            const formattedText = `${filteredText[0]}${filteredText[2] || 0}`;
-            if (text[0] >= 6) {
-                newMinute = `0${formattedText[1]}`;
-                newSelection = 2;
-            } else {
-                newMinute = `${formattedText[0]}${formattedText[1]}`;
-                newSelection = 1;
-            }
-        } else if (selectionMinute.start === 1) {
-            // Case when the cursor is at the second position.
-            // If we remove a value, prepend 0.
-            if (filteredText.length < 2) {
-                newMinute = `0${text}`;
+        } else if (selectionMinute.start === 1 && selectionMinute.end === 1) {
+            // The cursor is in-between the digits
+            if (trimmedText.length < 2) {
+                // We have removed the first digit. Replace it with 0 and move the cursor to the start.
+                newMinute = `0${trimmedText}`;
                 newSelection = 0;
-                setSelectionHour({start: 2, end: 2});
-                hourInputRef.current.focus();
             } else {
-                newMinute = `${text[0]}${text[1]}`;
+                newMinute = `${trimmedText[0]}${trimmedText[1] || 0}`;
                 newSelection = 2;
             }
-        } else if (filteredText.length < 2) {
-            // Case when the cursor is at the end and no text is selected.
-            newMinute = `${text}0`;
+        } else if (trimmedText.length <= 1 && trimmedText <= 5) {
+            /*
+             The trimmed text is from 0 to 5.
+             We are either replacing minutes with a single digit, or removing the last digit.
+             In both cases, we should append 0 to the remaining value.
+             Note: we must check the length of the filtered text to avoid incorrectly handling e.g. "01" as "1"
+            */
+            newMinute = `${trimmedText}0`;
             newSelection = 1;
+        } else {
+            newMinute = trimmedText;
+            newSelection = 2;
+        }
+
+        if (newMinute > 59) {
+            newMinute = minutes;
+        } else {
+            newMinute = newMinute.padStart(2, '0');
         }
 
-        setMinute(newMinute);
+        setMinutes(newMinute);
         setSelectionMinute({start: newSelection, end: newSelection});
+        if (newSelection === 0) {
+            focusHourInputOnLastCharacter();
+        }
```

Comment:
```suggestion
```

This is not helpful, it makes it hard to edit the first digit of the minutes

Reply:
It is implemented in `main`, I just moved it to the bottom. Will remove anyway 👌